### DISCUSSION TITLE
Deployed with gh-pages and now img tags aren't pulling images

### DISCUSSION THREAD

**Discussion by ethanbossen** [community/community#130787] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Not sure why the images aren't rendering in my project. Everything worked fine until I deployed to pages, now the images wont appear locally or on my gh-page set up. https://github.com/ethanbossen/Lemon-Reservations https://ethanbossen.github.io/Lemon-Reservations/ any help with this would be greatly appreciated!

**Comment by davevad93** [comment]
Hi @ethanbossen ,

I took a look at your repo and noticed that it's a React project, I think you need to have an `index.html` in the root of the repo and another one in the `public` folder. I've never actually deployed to GitHub Pages a React project but I remember a similar discussion from a couple of weeks ago, so take a look at this [thread](https://github.com/orgs/community/discussions/128670#discussioncomment-9822577) for better reference.

**Comment by Sicilianii** [comment]
First thing I noticed:
1. Your URL path to the page doesn't match the documentation... It should be ```https://<user>.github.io/<repository>/ ```
2. There is also a peculiarity on different project builders, you need to specify the home page in the configuration files, usually specify the path to the repository ```vite.config.ts```
```jsx
import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [react()],
  base: "https://<user>.github.io/<repository>/",
})
```

I recommend a detailed step by step documentation of GitHub Pages, keep the link:
```https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site```

If you find this as helpful, please mark this comment as an answer :)

**Comment by ava351** [comment]
I'd gladly assist you with the image issue you're encountering after deploying your project with GitHub Pages. Here are the common causes and solutions to troubleshoot the problem:

1. Incorrect File Paths:

Double-check file paths: Ensure the src attribute in your image tags points to the correct location of your images relative to your HTML file. Common mistakes include:
Using absolute paths (e.g., /images/my-image.jpg), which won't work on GitHub Pages.
Missing a leading slash for paths within the same directory (e.g., src="images/my-image.jpg" should be src="./images/my-image.jpg").
Use relative paths: Always use relative paths starting with a dot (.) to reference images within the same directory or subdirectories within your docs folder (the default deployment directory for GitHub Pages).

2. Case Sensitivity:

Match filenames: Make sure the filenames in your HTML and the actual image files match exactly (including case sensitivity). GitHub Pages is case-sensitive.
3. Image Uploads:

Verify image presence: Double-check that your images have been successfully uploaded to your GitHub repository. GitHub Pages will only serve files within the docs folder.
4. Clearing Browser Cache:

Hard reload: Sometimes, a cached version of your webpage might be causing the issue. Try a hard reload (Ctrl+Shift+R or Cmd+Shift+R) to force the browser to download the latest files from GitHub Pages.
5. Build Process (if applicable):

Review build configuration: If you're using a build process (like Webpack or Gulp), ensure it's configured correctly to handle image assets and not modify paths in a way that breaks them on GitHub Pages.
Additional Tips:

Test locally: Before deploying, test your project locally using a server like an HTTP server or a development server provided by your framework to ensure image paths work correctly.
Inspect browser console: Open your browser's developer tools (usually by pressing F12) and navigate to the console. Look for error messages related to image loading that might provide additional clues.
If none of these solutions work, please provide more context:

Share a snippet of your code, particularly the image tags and directory structure.
If you use a custom domain, mention that and any configuration details.
Mention any build tools or frameworks involved in your project.
By following these steps and providing additional information if needed, we can effectively diagnose and resolve the image loading issue on your GitHub Pages deployment.

**Comment by ethanbossen** [comment]
Figured it out, it was a pathing issue
